### ISSUE TITLE
goToDate not working properly

### ISSUE THREAD

**Issue by RimApp** [hoangnm/react-native-week-view#93]
Hello,

I try to use "goToDate" like the documentation says to update selectedDate prop.

I use "goToDate" prop to go to today date.
When I am :
-On the today date -1 day (11/05/2021) and less => OK. Navigate to the today date.
-On the today date (12/05/2021) => OK. Nothing happen.
-On the today date +1 day (13/05/2021) =>**KO. Nothing happen.**
-On the today date +2 days and more =>**KO. Navigate to the today date +1, so 13/05/2021.**

Have I forgotten something ?

Version that I use : 0.4.0.

```
       <WeekView
          ref={ref => {
            this.weekViewRef = ref;
          }}
          onEventPress={item => { // Navigate to my event form
            this.props.history.push({
              pathname: '/appointmentForm',
              state: {
                appointment: item.appointment,
              },
            });
          }}
          showTitle={false}
          showNowLine
          locale={GetCurrentLocale()} // Define my locale (here FR)
          headerStyle={styles.headerStyle}
          selectedDate={this.state.selectedDate}
          events={this.getAppointments()} // Get all my events
          numberOfDays={1}
          formatDateHeader={'ddd D'}
          hoursInDisplay={6}
          startHour={new Date().getHours()}
        />
```
When I press on the today button :
`this.weekViewRef.goToDate(new Date());`

**Comment by pdpino** [comment]
Hi @RimApp ,

Thanks for the detailed description. At first, it looks that there may be a of-by-one error on the `goToDate()` method.

If available, could you test what happens when you are in +3, +4 days? And -2, -3 days?

I will test this by myself soon, and let you know what I find.

**Comment by RimApp** [comment]
Hi @pdpino,

Thanks you for your quick reply.
+3, +4 days and the other superior days => **KO. Navigate to the today date+1, so 13/05/2021.**
-2, -3 days and the other inferior days => OK. Navigate to the today date, so 12/05/2021.

**Comment by pdpino** [comment]
@RimApp I fixed this and sent a PR :slightly_smiling_face: . It was a small issue in `goToDate()`

**Comment by pdpino** [comment]
This was fixed with PR #99 , that included changes from the original PR

**Comment by RimApp** [comment]
Thanks you its work :)
